spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/TestBaseWithCatalog.java

```diff
@@ -59,18 +86,30 @@ protected static Object[][] parameters() {
   }
 
   @BeforeAll
-  public static void createWarehouse() throws IOException {
+  public static void setUpAll() throws IOException {
     TestBaseWithCatalog.warehouse = File.createTempFile("warehouse", null);
     assertThat(warehouse.delete()).isTrue();
+    initRESTCatalog();
   }
 
   @AfterAll
-  public static void dropWarehouse() throws IOException {
+  public static void tearDownAll() throws Exception {
     if (warehouse != null && warehouse.exists()) {
       Path warehousePath = new Path(warehouse.getAbsolutePath());
       FileSystem fs = warehousePath.getFileSystem(hiveConf);
       assertThat(fs.delete(warehousePath, true)).as("Failed to delete " + warehousePath).isTrue();
     }
+    stopRESTCatalog();
+  }
+
+  private static void initRESTCatalog() {
+    restCatalog = RCKUtils.initCatalogClient(REST_SERVER_EXTENSION.config());
```

Comment:
So rather than making the client like theis and making the RESTUtils public.  I think you can actually create the client in the extension and just grab it out of the context:

```java
// In the RESTServerExtension class
@Override
public void beforeAll(ExtensionContext extensionContext) throws Exception {
   // ... other init
   // store the client
    Store store = context.getStore(Namespace.GLOBAL);
    store.put("rest-client", RCKUtils.initCatalogClient(...));
  }
}
```

```java
    @Override
    public void beforeEach(ExtensionContext context) {
        // get the rest catalog client from the context store
        Store store = context.getStore(Namespace.GLOBAL);
        this.restCatalog = store.get("rest-client");
    }
```

Reply:
I made `RCKUtils` package protected again in the latest commit. A little bit different from what you suggested above is: the Spark integ test rely on REST client being initialized in static context, so initializing restCatalog in `beforeEach` might be hard. Solved the problem by making `RESTServerExtention` taking control over client life cycle along side server life cycle. Anyway, in the end, `RCKUtils` is package protected again.